File: scripts/debug-sandbox.py
#!/usr/bin/env python3
"""
Debug script to test Sandbox execution.
"""

import subprocess
import sys
import os
import tempfile

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

def wrap_code(code: str) -> str:
    """Wrap user code for Sandbox execution."""
    return f"""
import asyncio
import sys
import os

# Add current directory to Python path
if os.getcwd() not in sys.path:
    sys.path.insert(0, os.getcwd())

from core.skill.executor import SkillExecutor

async def main():
    executor = SkillExecutor()
    try:
{chr(10).join('        ' + line for line in code.split(chr(10)))}
    except Exception as e:
        print(f"Error: {{e}}")
        import traceback
        traceback.print_exc()

asyncio.run(main())
"""

def main():
    print("=== Sandbox Debug Test ===\n")

    print("Environment:")
    print(f"  Python: {sys.executable}")
    print(f"  CWD: {os.getcwd()}")
    print(f"  PYTHONPATH: {os.environ.get('PYTHONPATH', 'Not set')}")
    print("")

    # Test 1: Simple code
    print("Test 1: Simple Python code")
    code1 = "print('Hello from Python')"
    wrapped1 = wrap_code(code1)

    with tempfile.NamedTemporaryFile(mode='w', suffix='.py', delete=False) as f:
        f.write(wrapped1)
        temp1 = f.name

    try:
        result = subprocess.run(
            [sys.executable, temp1],
            capture_output=True,
            text=True,
            timeout=5,
            env={**os.environ, 'PYTHONPATH': os.getcwd()}
        )

        print(f"  Success: {result.returncode == 0}")
        print(f"  Output: {result.stdout.strip()}")
        if result.stderr:
            print(f"  Stderr: {result.stderr.strip()}")
    finally:
        os.unlink(temp1)

    print("")

    # Test 2: Import SkillExecutor
    print("Test 2: SkillExecutor already imported")
    code2 = "print('SUCCESS: SkillExecutor is accessible')"
    wrapped2 = wrap_code(code2)

    with tempfile.NamedTemporaryFile(mode='w', suffix='.py', delete=False) as f:
        f.write(wrapped2)
        temp2 = f.name

    try:
        result = subprocess.run(
            [sys.executable, temp2],
            capture_output=True,
            text=True,
            timeout=10,
            env={**os.environ, 'PYTHONPATH': os.getcwd()}
        )

        print(f"  Success: {result.returncode == 0}")
        print(f"  Output: {result.stdout.strip()}")
        if result.stderr:
            print(f"  Stderr: {result.stderr.strip()}")
    finally:
        os.unlink(temp2)

    print("")

    # Test 3: Skill Registry
    print("Test 3: Skill Registry")
    code3 = """
from core.skill.registry import SkillRegistry
registry = SkillRegistry()
skills = await registry.scan()
print(f"SUCCESS: Found {len(skills)} skills")
for name in skills.keys():
    print(f"  - {name}")
"""
    wrapped3 = wrap_code(code3)

    with tempfile.NamedTemporaryFile(mode='w', suffix='.py', delete=False) as f:
        f.write(wrapped3)
        temp3 = f.name

    try:
        result = subprocess.run(
            [sys.executable, temp3],
            capture_output=True,
            text=True,
            timeout=10,
            env={**os.environ, 'PYTHONPATH': os.getcwd()}
        )

        print(f"  Success: {result.returncode == 0}")
        print(f"  Output: {result.stdout.strip()}")
        if result.stderr:
            print(f"  Stderr: {result.stderr.strip()}")
    finally:
        os.unlink(temp3)

    print("")

if __name__ == "__main__":
    main()

File: scripts/debug-sandbox.ts
/**
 * Debug script to test Sandbox execution
 */

import { SandboxFactory } from '../src/core/sandbox/factory';

async function main() {
  console.log('=== Sandbox Debug Test ===\n');

  console.log('Environment:');
  console.log('  PYTHON_PATH:', process.env.PYTHON_PATH);
  console.log('  PYTHONPATH:', process.env.PYTHONPATH);
  console.log('  CWD:', process.cwd());
  console.log('');

  // Create sandbox
  const sandbox = SandboxFactory.create({
    type: 'local',
    config: {
      pythonPath: process.env.PYTHON_PATH || 'python3',
      timeout: 30000,
    },
  });

  console.log('Sandbox created:', sandbox.getInfo());
  console.log('');

  // Test 1: Simple Python code
  console.log('Test 1: Simple Python code');
  const test1 = await sandbox.execute('print("Hello from Python")', {
    skills: [],
    skillImplPath: process.cwd(),
    sessionId: 'debug-test-1',
    timeout: 5000,
  });
  console.log('  Success:', test1.success);
  console.log('  Output:', test1.stdout);
  console.log('  Stderr:', test1.stderr);
  console.log('  Error:', test1.error);
  console.log('');

  // Test 2: Import SkillExecutor
  console.log('Test 2: Import SkillExecutor');
  const test2Code = `
import sys
print('Python path:', sys.path[:3])
try:
    from core.skill.executor import SkillExecutor
    print('SUCCESS: SkillExecutor imported')
except Exception as e:
    print(f'ERROR: {e}')
    import traceback
    traceback.print_exc()
`;

  const test2 = await sandbox.execute(test2Code, {
    skills: [],
    skillImplPath: process.cwd(),
    sessionId: 'debug-test-2',
    timeout: 10000,
  });
  console.log('  Success:', test2.success);
  console.log('  Output:', test2.stdout);
  console.log('  Stderr:', test2.stderr);
  console.log('  Error:', test2.error);
  console.log('');

  // Cleanup
  await sandbox.cleanup();
}

main().catch(console.error);
